### DISCUSSION TITLE
How to handle Barebone Server

### DISCUSSION THREAD

**Discussion by MaKaNu** [netbox-community/netbox#15354] [Help Wanted!] [answered]
Since Barebone servers like the Supermicro CSE-825TQC-R1K03LPB could fit different mainboards the interface could differ from server to server. Are Mainboards devices as well and I create a parent children dependency?

**Comment by candlerb** [comment, marked as answer]
Well, you *could* do it that way, where the chassis is a dummy parent device that holds the child device (motherboard), which is the "real" device. This has some downsides in the UI, where you don't immediately see the child device in rack elevations, and you have to open the child device to see its interface connections.

You could also do it using modules: in this case the chassis is a device with a "module bay", and the motherboard is a "module type" which can have its own set of interfaces. I think that's somewhat better, as there's only one "device" and all the interfaces get copied to it when the module is installed.

But personally, I'd be inclined to take the simpler option of making a different Device Type for each variation of this chassis that you use (with a particular motherboard installed).

**Reply by MaKaNu** [reply]
As I was creating the module bay I realized that also a device bay exists. But I think this is better used for blade devices?

**Reply by candlerb** [reply]
Device bay is the implementation of parent/child devices. The device bay exists in the parent device, and the child device is itself an independent device, but with the Parent/Child status set to "child". The child fits into a device bay of the parent.

Device bay is for when the parent is "dumb" and each child is an independently manageable device in its own right. You are correct: blade servers are a good example of this.  Another example might be when you have two half-width devices which fit in a single 1U slot: the mounting bracket is the parent device (which takes up 1U of space) and the half-width devices are the children.  I've also used this for modular patch panel frames.

Module bay is where you have just one device, which has its own management IP, but there are slots that expand that device and which may add components (interfaces etc).  Router line cards are a good example of this.

For things you add that don't have any interfaces, then Inventory Items are simpler. Things like DIMMs and hard drives are good examples here.

**Reply by MaKaNu** [reply]
I like the approach with the module bay and it fits our needs perfect. Since it seems modules themself couldn't themself have module bays how should I approach Extension Cards? Create Bays only on device level if needed?

**Reply by candlerb** [reply]
You'd have to put more module bays on the top-level Device.  This means that, for example, PCI cards would appear to plug into the chassis, not the motherboard.

As I said before: the simplest option is to create a Device Type that represents the combination of chassis + motherboard.

**Reply by MaKaNu** [reply]
Ok I was afraid this would be the answer.
